examples/federated-average/run.py

```diff
@@ -20,36 +20,34 @@
     config = tfe.LocalConfig(player_names_fixed + player_names_inputter)
 
 
-class Inputter(tfe.io.InputProvider):
-    def provide_input(self) -> List[tf.Tensor]:
-        # pick random tensor to be averaged
-        return [tf.random_normal(shape=(10,))]
+def provide_input() -> List[tf.Tensor]:
+    # pick random tensor to be averaged
+    return [tf.random_normal(shape=(10,))]
 
 
-class ResultReceiver(tfe.io.OutputReceiver):
-    def receive_output(self, tensors: List[tf.Tensor]) -> tf.Operation:
-        average, = tensors
-        # simply print average
-        return tf.Print([], [average], summarize=10, message="Average:")
+def receive_output(*args: List[tf.Tensor]) -> tf.Operation:
+    average, = args
+    # simply print average
+    return tf.Print([], [average], summarize=10, message="Average:")
 
 
 # create players based on names from above
-inputters = [Inputter(config.get_player(name)) for name in player_names_inputter]
-result_receiver = ResultReceiver(config.get_player('result_receiver'))
 server0 = config.get_player('server0')
 server1 = config.get_player('server1')
 crypto_producer = config.get_player('crypto_producer')
 
 with tfe.protocol.Pond(server0, server1, crypto_producer) as prot:
 
     # get input from inputters as private values
-    inputs = [prot.define_private_input(inputter)[0] for inputter in inputters]
+    inputs = [
+        prot.define_private_input(config.get_player(name), provide_input)[0] for name in player_names_inputter
```

Comment:
you can avoid the `[0]` here by returning a `tf.Tensor` from `provide_input` instead of a list